### PR TITLE
Fish 6611 - reporter to summarize the test results

### PR DESCRIPTION
**Pull request by simonladen**
Add summarizer to the tck runner. 
It takes up to 4 arguments to run:
`"<tck report file> <format:jUnitReport|summaryTxt|testSet|failsafeSummary> [testSuiteName] [<output report path>]"`
See the test samples for the different formats handled.
It runs, in Unix, with a command like this:
mvn exec:java -Dexec.arguments=[**/*junit-report.xml,jUnitReport,testSuiteName]
(because of the file separator, this command fails in Windows)
It produces a text file: /summarizer/results/summary-$testSuiteName-$timestamp.txt 

the summary itself looks like this:
### testSuiteName

Completed running 1713 tests
Number of tests failed 9
Number of tests with errors 2
Number of tests skipped 1

### CODE REVIEW

File: summarizer/pom.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>fish.payara.internal</groupId>
+    <artifactId>tck-summarizer</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (aubi) [inline summarizer/pom.xml:52]: Why Java 8? I suggest 11 as Jakarta 10 depends on 11 anyway.

File: summarizer/src/main/java/fish/payara/internal/tcksummarizer/TxtParser.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                  // pass the path to the file as a parameter
+                  File file = new File(inputFilePath);
+                  Scanner sc = new Scanner(file);
+                  
+                  while (sc.hasNextLine()) 
+                  {
+                      String currentLine = sc.nextLine();
+                      
+                      if (currentLine.contains("Tests run: "))
+                      {
```
Comment (aubi) [inline summarizer/src/main/java/fish/payara/internal/tcksummarizer/TxtParser.java:114]: For new files, I suggest format them (Java Coding Standard says opening brace on the same line).

File: summarizer/src/main/java/fish/payara/internal/tcksummarizer/XmlParser.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
+/*
+ *    DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ *    Copyright (c) [2019] Payara Foundation and/or its affiliates. All rights reserved.
```
Comment (aubi) [inline summarizer/src/main/java/fish/payara/internal/tcksummarizer/XmlParser.java:4]: It's 2022 :-)

**Review by aubi — CHANGES_REQUESTED** [review]

**Comment by aubi** [comment]
Could you add a README.md with the description?

**Comment by aubi** [comment]
I was trying `beanvalidation-2.0-junit-report.xml` and the summarizer worked well. It's just important to note in the readme:
Processes file inside current dir, file must be entered as `./beanvalidation-2.0-junit-report.xml` and it processes just one file.
So after fixing the minor issues, the PR can be approved.

File: summarizer/src/main/java/fish/payara/internal/tcksummarizer/XmlParser.java (hunk 3 above)
Comment (simonladen) [reply summarizer/src/main/java/fish/payara/internal/tcksummarizer/XmlParser.java:4]: Oops, indeed! The joy of copy/pasting

File: summarizer/pom.xml (hunk 1 above)
Comment (simonladen) [reply summarizer/pom.xml:52]: Indeed, I'll update that

**Review by aubi — APPROVED** [review]
LGTM

File: summarizer/README.md [hunk 4]
```diff
@@ -24,12 +24,29 @@ Run maven test from summarizer
...
+>    mvn exec:java -Dexec.arguments=[**/*junit-report.xml,jUnitReport,testSuiteName]
+
+###  Alternative ways to execute the summarizer:
+
+#### Running the jar 
+
+Run the following command to create the jar in ./target:
+
+>   mvn package
+
```
Comment (aubi) [inline summarizer/README.md:43]: Remove this, everybody knows, how to build a project.

**Review by aubi — CHANGES_REQUESTED** [review]

**Pull request merged**